### PR TITLE
feat(smart-contracts): add Sepolia testnet

### LINKED ISSUE #11654: Add Sepolia
Goerli will in fact go away [at some point in 2023](https://ethereum.org/en/developers/docs/networks/#goerli)... so let's add Sepolia!

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds deployment infor and config for sepolia testnet

- [x] deploy contracts
- [x] deploy subgraph
  - [x] sync graph
  - [x] publish graph
- [x] verify proxy ~with sourcify~

# Issues

Fixes #11654
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
finally managed to verify Lock Proxy on Sepolia using this method https://github.com/unlock-protocol/unlock/issues/12578#issuecomment-1688086026

File: packages/networks/src/networks/sepolia.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { NetworkConfig } from '@unlock-protocol/types'
+
+export const sepolia: NetworkConfig = {
+  blockScan: {
+    url: (address: string) => `https://blockscan.com/address/${address}`,
+  },
+  chain: 'base',
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:7]:
```suggestion
  chain: 'sepolia',
```

File: packages/networks/src/networks/sepolia.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    urls: {
+      address: (address: string) =>
+        `https://sepolia.etherscan.io/address/${address}`,
+      base: `https://sepolia.etherscan.io/`,
+      token: (address: string, holder: string) =>
+        `https://sepolia.etherscan.io/token/${address}?a=${holder}`,
+      transaction: (hash: string) => `https://sepolia.etherscan.io/tx/${hash}`,
+    },
+  },
+  featured: true,
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:21]:
```suggestion
  featured: false,
```

Let's set to `false` first so we can test everything works as expected

File: packages/networks/src/networks/sepolia.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  fullySubsidizedGas: false,
+  hooks: {
+    onKeyPurchaseHook: [],
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:24]: Not for this PR butbefore we set `feature` to `true`, let's make sure we deploy the same hooks as on the other chains...

File: packages/networks/src/networks/sepolia.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  },
+  id: 11155111,
+  isTestNetwork: true,
+  keyManagerAddress: '',
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:28]: Same here!

File: packages/networks/src/networks/sepolia.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  nativeCurrency: {
+    coingecko: 'ethereum',
+    decimals: 18,
+    name: 'ETH',
+    symbol: 'ETH',
+  },
+
+  opensea: {
+    collectionUrl: (lockAddress) =>
+      `https://testnets.opensea.io/assets/base/${lockAddress}`,
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:41]:
```suggestion
      `https://testnets.opensea.io/assets/sepolia/${lockAddress}`,
```

File: packages/networks/src/networks/sepolia.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    tokenUrl: (_lockAddress, _tokenId) =>
+      `https://testnets.opensea.io/assets/base/${_lockAddress}/${_tokenId}`,
```
Comment (julien51) [inline packages/networks/src/networks/sepolia.ts:43]:
```suggestion
      `https://testnets.opensea.io/assets/sepolia/${_lockAddress}/${_tokenId}`,
```

**Review by julien51 — COMMENTED** [review]
few small changes and then LG!

**Review by julien51 — APPROVED** [review]

**Pull request merged**